### ISSUE TITLE
Tracer transport simulation fails out-of-box for dev/14.2.0

### ISSUE THREAD

**Issue by sdeastham** [geoschem/gchp#331]
### Name and Institution (Required)

Name: Seb Eastham
Institution: MIT

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://gchp.readthedocs.io/en/latest/reference/SUPPORT.html)
- [x] [User manuals](https://gchp.readthedocs.io/en/latest/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://gchp.readthedocs.io/en/latest/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/GCHP/issues)

### Description of your issue or question

When running the GCHP tracer transport simulation out of the box (branch dev/14.2.0), the simulation fails. The following message is printed in the gchp.20190101_0000z.log file, specifically with verbose set to `true` in the `HEMCO_Config.rc` file:

``` Contents of always-to-read list:
HEMCO: Leaving SetReadList (HCO_CONFIG_MOD.F90) ( 2)
HEMCO: Leaving HCO_INIT (HCO_DRIVER_MOD.F90) ( 1)
HEMCO: Entering HCOX_INIT (HCOX_DRIVER_MOD.F90) ( 1)
The Volcano extension is turned off.
HEMCO: Entering HCOX_GC_RNPBBE_INIT (HCOX_GC_RNPBBE_MOD.F90) ( 2)
-------------------------------------------------------------------------------
Use the following species (Name: HcoID):
 Rn222:           1
 Be7:           4
 Be7s:           5
 Be10:           6
 Be10s:           7
HEMCO: Leaving HCOX_GC_RNPBBE_INIT (HCOX_GC_RNPBBE_MOD.F90) ( 2)
HEMCO: Leaving HCOX_INIT (HCOX_DRIVER_MOD.F90) ( 1)

HEMCO ERROR [0005]:  Cannot find extension Nr.         -999
 --> LOCATION: GetExtSpcStr (hco_extlist_mod.F90)

HEMCO ERROR [0005]: ERROR 24
 --> LOCATION: HCO_GetExtHcoID (hco_state_mod.F90)

HEMCO ERROR [0005]: ERROR 2
 --> LOCATION: HCOX_VOLCANO_INIT (HCOX_VOLCANO_MOD.F90)

HEMCO ERROR [0005]: Error encountered in "HCOX_Volcano_Init"!
 --> LOCATION:  -> at HCOX_INIT (in module HEMCO/Extensions/hcox_driver_mod.F90

GEOS-Chem ERROR [0005]: Error encountered in "HCOX_Init"!
 --> LOCATION:  -> at HCOI_GC_Init (in module GeosCore/hco_interface_gc_mod.F90)

HEMCO ERROR [0004]:  Cannot find extension Nr.         -999
 --> LOCATION: GetExtSpcStr (hco_extlist_mod.F90)

HEMCO ERROR [0004]: ERROR 24
 --> LOCATION: HCO_GetExtHcoID (hco_state_mod.F90)

HEMCO ERROR [0004]: ERROR 2
 --> LOCATION: HCOX_VOLCANO_INIT (HCOX_VOLCANO_MOD.F90)

HEMCO ERROR [0004]: Error encountered in "HCOX_Volcano_Init"!
 --> LOCATION:  -> at HCOX_INIT (in module HEMCO/Extensions/hcox_driver_mod.F90

GEOS-Chem ERROR [0004]: Error encountered in "HCOX_Init"!
 --> LOCATION:  -> at HCOI_GC_Init (in module GeosCore/hco_interface_gc_mod.F90)
```
No modifications have been made to the directory. I am vaguely aware that extension -999 is a catch-all of some kind in HEMCO but beyond that have no idea what could be causing this. I have verified that a conventional, full-chemistry simulation does not have this issue (it does fail, but because of a missing entry for SPC_BUTDI in the restart?).

**Comment by lizziel** [comment]
I can help with your full chemistry issue. Standard simulations fail if a species is missing from the restart file (request from GCSC). We do not have spun-up restart files for 14.2.0 yet so the 14.0 restart file will have missing species. I disabled this requirement for transport tracers in GCHP 14.2.0 but not for standard simulation. To allow missing species look for the toggle in `setCommonRunSettings.sh` that is called `Require_Species_in_Restart`.

**Comment by sdeastham** [comment]
Thanks @lizziel ! I can confirm that that does fix the fullchem simulation. It's running merrily at C24 using MERRA-2 for 2019-07-01 (although I'm seeing an alarming number of integrator errors). Which unfortunately leaves us with the open question of what is going on in tracer transport. @msulprizio - did you have any issues running the GCHP tracer transport tests for 14.2.0 or are they not started yet?

**Comment by msulprizio** [comment]
@sdeastham The transport tracer simulations have been passing GCHP integration tests (1-hour simulations at c24). I have not fully validated the GCHP transport tracer simulation yet, but I am working on that today.

The error that you're getting seems strange -- the transport tracer simulation should not be attempting to use the volcano extension. I will try to reproduce this on my end.

**Comment by lizziel** [comment]
@msulprizio was not able to reproduce the issue. @sdeastham, could you submit your config files? Also report the last commit of your geos-chem and hemco repos.

**Comment by sdeastham** [comment]
Sure! Please let me know if I missed anything critical - see attached.
[config_files_and_logs.zip](https://github.com/geoschem/GCHP/files/12113264/config_files_and_logs.zip)

EDIT: I've also reached out to @christophkeller in case they have any additional insights.

**Comment by lizziel** [comment]
I think this is a bug that came in with a recent update to suppress log write unless root. See [here](https://github.com/geoschem/HEMCO/blob/1ecfc19e8e7382912f469a5511af7e3012d2fdd8/src/Extensions/hcox_volcano_mod.F90#L358-L373). It should return if the extension number is negative. However, with the update it is only returning if root. The model is then crashing on non-root threads.

I am not sure why our integration tests did not catch this. Still investigating. @sdeastham, if you make that return not on root it should fix your issue.

**Comment by lizziel** [comment]
See fix [here](https://github.com/geoschem/HEMCO/pull/230). I'll run integration tests before making the PR ready to review/merge. I should have done that with the update that caused the bug (it was from me).

**Comment by sdeastham** [comment]
Thanks @lizziel - I can confirm that this fixes the issue!

**Comment by msulprizio** [comment]
The [fix](https://github.com/geoschem/HEMCO/pull/230) has now been merged into the HEMCO 3.7.0 development branch. I will close out this issue.
